coins/ethereum/src/contract.rs

```diff
@@ -0,0 +1,127 @@
+use ethers::{contract::ContractFactory, prelude::*};
+use eyre::{eyre, Result};
+use serai_processor::coin::ethereum;
+use std::fs::File;
+use std::sync::Arc;
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum EthereumError {
+    #[error("failed to verify Schnorr signature")]
+    VerificationError,
+}
+
+abigen!(
+    Schnorr,
+    "./schnorr-verify/artifacts/contracts/Schnorr.sol/Schnorr.json",
+    event_derives(serde::Deserialize, serde::Serialize),
+);
+
+pub async fn deploy_schnorr_verifier_contract(
+    client: Arc<SignerMiddleware<Provider<Http>, LocalWallet>>,
+) -> Result<schnorr_mod::Schnorr<SignerMiddleware<Provider<Http>, LocalWallet>>> {
+    let path = "./schnorr-verify/artifacts/contracts/Schnorr.sol/Schnorr.json";
+    let artifact: HardhatArtifact = serde_json::from_reader(File::open(path).unwrap()).unwrap();
```

Comment:
This can be read at compile time.

Also, pub(crate) since this should be limited to tests/a `src/bin` app.

Reply:
if i move the tests to a new folder it complains that `deploy_schnorr_verifier_contract` if i use `pub(crate)` :/

Reply:
Uhhhh `src/tests`, and then you still do `#[cfg(test)] mod test;`